Pull request to branch main

Georg35-hash opened the pull request.

GertSer committed:
The first commit of the whole project

GertSer committed:
Added new section, fixed js code

GertSer committed:
Finished section offer

GertSer committed:
Renamed some HTML classes, fixed colors

GertSer committed:
Fixed js

GertSer committed:
Added new section, js code. Fixed the cinematic header

GertSer committed:
Added new subscribe section, needs some work

GertSer committed:
Added footer

GertSer committed:
Added shadow to header and hero section, shadows need optimization

GertSer committed:
Fixed styles + added shadows

GertSer committed:
Begin of responsive lay out

GertSer committed:
Added breakpoint 768px, needs some work

GertSer committed:
Fixed overflow: hidden, and some bugs

GertSer committed:
Added new breakpoint

GertSer committed:
Fixed header for responsive

GertSer committed:
Fixed breakpoint 576px

GertSer committed:
Edited size of catalogue

GertSer committed:
For phones

GertSer committed:
One more edit

GertSer committed:
One more edit

GertSer committed:
And one more

GertSer committed:
Top menu

GertSer committed:
Fixed footer for 768px + navi arrows in section blog

GertSer committed:
Footer for 576px

GertSer committed:
Fixed buttons animation + subscribe section

GertSer committed:
Shadows need optimization

GertSer committed:
Fixed 576px section catalogue

GertSer committed:
Fixed offer section

GertSer committed:
Final fix of all bugs

GertSer committed:
Fixed bugs

GertSer committed:
Added 1024px, 320px sections

GertSer committed:
Fixed footer and some bugs

GertSer committed:
Added new breakpoin 425px + fixed old bugs

GertSer committed:
Fixed some bugs

GertSer committed:
Added fixed postition for header

GertSer committed:
Added default state .activr for sidebar

GertSer committed:
Added modal windows

GertSer committed:
Responsive modal window

GertSer committed:
Fixed size of window

GertSer committed:
Fixed footer

GertSer committed:
Fixed adaptive width

GertSer committed:
Added basic animation + one more scss file for animations

GertSer committed:
Optimized Shadows

GertSer committed:
Fixied bugs with slider-navi.js. Separated scripts

ablyazov-dariy commented on package.json:
```diff
@@ -0,0 +1,16 @@
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
remove unsed scripts

Georg35-hash replied:
Done

ablyazov-dariy commented on package.json:
```diff
@@ -0,0 +1,16 @@
+  "license": "ISC",
+  "description": "",
+  "dependencies": {
+    "sass": "^1.83.0"
```
sass should be a development dependency (devDependencies)

Georg35-hash replied:
Done

ablyazov-dariy commented on src/assets/catalogue/bed sheets.svg:
Move assets outside of src for better separation of source code and static assets.

Avoid spaces in file names; use hyphens or underscores instead (e.g., bed-sheets.svg or bed_sheets.svg).

Georg35-hash replied:
Done

ablyazov-dariy commented on src/dist/style.css:
Remove the dist directory from within src. Compiled output like style.css should not be stored in your source directory. Add dist/ to your .gitignore file to prevent it from being committed to version control.

Georg35-hash replied:
Moved dist out of src. Done

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="./src/dist/style.css">
+    <title>Bed Linen Online</title>
```
add a favicon

Georg35-hash replied:
Done

ablyazov-dariy commented on index.html:
Please ensure your code is consistently formatted. Consider using a formatter like Prettier or your IDE's built-in formatting tools.

Georg35-hash replied:
Installed Prettier. Done

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+                        They offer several color options, patterns, and textures so that you can create the perfect
+                        ambience in your bedroom. Their quality is second to none, with materials well-
+                    </p>
+                    <a href="#" class="sleepy-shop__link animate-on-scroll" data-content="<h2>Sleepy Shop</h2><p>Sleepy is the perfect bedding store for anyone looking to give their bedroom a makeover.
```
The use of inline HTML within the data-content attribute looks messy and is not scalable. Please consider using the native <dialog> element, a <template> element, or a JavaScript-driven approach to manage this content. Storing large amounts of HTML directly in attributes is not maintainable, especially if the content grows.

Georg35-hash replied:
Done, i did it with variables in script

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+                                    <div class="product" bis_skin_checked="1">
+                                        <p>Pillow 1</p>
+                                        <span>$120 <button><img class="product_icon"
+                                                    src="./src/assets/catalogue/sale_icon.svg" alt="sale_icon"></button<
```
Closing tag is not valid

Georg35-hash replied:
Done

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+                <img class="hero__img animate-on-scroll" src="./src/assets/hero/Hero-pic.svg" alt="sofa">
+                <img class="hero__background animate-on-scroll" src="./src/assets/hero/feather-bg.svg" alt="feather">
+                <img class="hero__shadow" src="./src/assets/hero/hero-shadow.png">
+                <img class="hero__light" src="./src/assets/hero/hero-light.png">
```
Missing required 'alt' attribute

Georg35-hash replied:
Done

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+                <form class="subscribe-form">
+                    <div class="input-container">
+                        <img src="./src/assets/subscribe/email.svg" class="input-icon" alt="input-icon" />
+                        <input type="email" placeholder="Enter your email" class="subscribe-input" required />
```
Missing associated label

Georg35-hash replied:
Added

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+                <ul class="header__icons-list-mobile">
+                    <li class="header__icons-item-mobile">
+                        <a href="#">
+                            <img src="./src/assets/header//shopping.svg" alt="shopping">
```
path

Georg35-hash replied:
Done

ablyazov-dariy commented on modal-handler.js:
Please organize all .js files within a dedicated directory, such as src/scripts or src/js.

Georg35-hash replied:
Organized

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+            <p>Copyright 2023 Sleepy Shop. All Rights Reserved.</p>
+        </div>
+    </footer>
+    <div class="modal" id="modal">
```
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

Georg35-hash replied:
I have some bugs with tag <template>

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,490 @@
+            <div class="modal-text"></div>
+        </div>
+    </div>
+    <script src="slider-navi.js"></script>
```
Please consider using a module bundler (like Vite, Webpack, Parcel, or Rollup) to manage these script files

Georg35-hash replied:
Done. Use Vite for module bundler

ablyazov-dariy commented on modal-handler.js:
```diff
@@ -0,0 +1,37 @@
+function initModal() {
+    const modal = document.querySelector(".modal");
+    const modalText = document.querySelector(".modal-text");
+    const closeBtn = document.querySelector(".close-btn");
```
To improve efficiency, consider using modal.querySelector instead of document.querySelector for elements within the modal. This will scope your queries to the modal element itself.

Georg35-hash replied:
Done

ablyazov-dariy commented on modal-handler.js:
```diff
@@ -0,0 +1,37 @@
+    const modalText = document.querySelector(".modal-text");
+    const closeBtn = document.querySelector(".close-btn");
+
+    if (!modal || !modalText) return;
```
Instead of checking for null and returning, consider removing this check and allowing the script to throw an error if modal or modalText is not found. This will make debugging easier by providing a clear error message.

Georg35-hash replied:
Done

ablyazov-dariy commented on modal-handler.js:
```diff
@@ -0,0 +1,37 @@
+        btn.addEventListener("click", (e) => {
+            e.preventDefault();
+            const content = btn.getAttribute("data-content");
+            modalText.innerHTML = content;
```
Consider extracting this modal-opening logic into a separate function, like `openModal(content)`. This will improve code organization and reusability.

Georg35-hash replied:
Done

ablyazov-dariy commented on modal-handler.js:
Please ensure that the file name and the primary function or class defined within it are consistently named. For example, if you have a function named initModal, the file should be named init-modal.js. 

> "To improve code organization and follow the Single Responsibility Principle (SRP), consider separating the modal-related logic into smaller, more focused functions. Instead of a single wrapper function, you could have functions like:
>
> * `openModal(content)`: Handles opening the modal and setting its content.
> * `closeModal()`: Handles closing the modal.
> * `addModalEventListeners()`: Attaches event listeners to the modal elements.
>
> This approach will make your code more modular, easier to understand, and simpler to maintain."

Georg35-hash replied:
Fixed

ablyazov-dariy commented on scroll.js:
```diff
@@ -0,0 +1,24 @@
+
+    elements.forEach(el => observer.observe(el));
+}
+
```
vertical spacing might be a bit too big
2 lines is generally enough

Georg35-hash replied:
Fixed

ablyazov-dariy commented on scroll.js:
Since initScrollAnimations is not being exported or reused, consider removing the wrapper function and directly executing the code within it. This will simplify the structure.

ablyazov-dariy commented on src/styles/main.scss:
```diff
@@ -0,0 +1,13 @@
+@use './subscribe';
+@use './footer';
+@use './global';
+@use './animation'
```
;

Georg35-hash replied:
Fixed

ablyazov-dariy commented on src/styles/_subscribe.scss:
Use a code formatter like Prettier or your IDE's built-in formatting tools to ensure consistent code style.
 
 Avoid overly specific selectors. While nesting can be useful, excessive nesting can lead to selectors that are difficult to override and maintain. Aim for a balance between nesting and specificity.

Georg35-hash replied:
I will take note. Thx

ablyazov-dariy commented on src/styles/_reset.scss:
Instead of a separate file, consider placing this CSS reset directly in main.scss. It's a small reset and doesn't warrant a dedicated file.

Georg35-hash replied:
Done

ablyazov-dariy commented on src/styles/_blog.scss:
```diff
@@ -0,0 +1,394 @@
+.blog-slider {
+    color: #F5F7FA;
+    font-family: "Cabinet-Grotesk", sans-serif;
+    margin: 98px 0px 0px 40px;
```
The unit of measure 'px' is redundant when setting a value to 0. For example, 0px can simply be 0.  I noticed this in several places throughout the code.

Georg35-hash replied:
Fixed

ablyazov-dariy commented on src/styles/_catalogue.scss:
```diff
@@ -0,0 +1,713 @@
+                            filter: blur(2px);
+                            width: 226.45px;
+                            cursor: pointer;
+                            width: 100%;
```
Property width overwrites property width

Georg35-hash replied:
Fixed

ablyazov-dariy commented on src/styles/_catalogue.scss:
```diff
@@ -0,0 +1,713 @@
+
+                        img {
+                            filter: blur(2px);
+                            width: 226.45px;
```
Using fractional pixel values like 226.45px can lead to inconsistent rendering across browsers

Georg35-hash replied:
Fixed

GertSer committed:
Deleted unsed scripts, fixed bug with button

Georg35-hash committed:
Updated .gitignore

GertSer committed:
Fixed bugs/issues from PR

GertSer committed:
Merge branch 'dev' of github.com:Georg35-hash/Bed-Linen-Shop into dev

GertSer committed:
Fixed nesting + parallax

Georg35-hash commented:
Bugs/issues have been fixed, and all comments have been addressed

GertSer committed:
Fixed parallax logic + refactored scss

ablyazov-dariy commented on index.html:
```diff
@@ -0,0 +1,638 @@
+                        alt="Pillow 3"
+                      />
+                    </label>
+                    <div class="product" bis_skin_checked="1">
```
Custom attributes starts with data-

ablyazov-dariy commented on index.html:
Use https://validator.w3.org/ to validate your HTML.

ablyazov-dariy commented on package.json:
```diff
@@ -0,0 +1,19 @@
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview",
+    "sass": "sass --watch src/main.scss dist/style.css"
```
Are you still using this script?

ablyazov-dariy commented on public/assets/blog/less.svg:
The path 'public/assets/blog/less.svg' is redundant. We should standardize either 'public' or 'assets' as the root for static assets to avoid confusion. Consider using 'assets/blog/less.svg' or '/blog/less.svg' (if 'public' serves as the root).

Asset path 'public/assets/blog/less.svg' should be an absolute path: '/assets/blog/less.svg'. This allows us to effectively utilize the <base> tag, ensuring consistent asset loading regardless of the current page's URL depth.
We will need this for hosting on gh pages

ablyazov-dariy commented on src/styles/_reset.scss:
🛰️

ablyazov-dariy replied:
Seriously, remove this

ablyazov-dariy commented on src/styles/_offer.scss:
```diff
@@ -0,0 +1,245 @@
+    }
+  }
+
+  /* Existing 320px Breakpoint */
```
Is this a ChatGPT comment?
Do we really need it?
If we need it, at least be consistent and add them to every file or remove them everywhere

ablyazov-dariy commented on src/styles/_header.scss:
```diff
@@ -0,0 +1,238 @@
+  }
+}
+
+/* Media */
```
Media what? 
Remove this comment; it doesn't help understand the code and, therefore, should be removed

ablyazov-dariy commented on src/styles/_global.scss:
```diff
@@ -0,0 +1,136 @@
+  scroll-behavior: smooth;
+}
+body::-webkit-scrollbar {
+  display: none; /* Chrome, Safari */
```
Are we only supporting Chrome and Safari, or do we remove the scroll bar just for those 2

ablyazov-dariy commented on src/styles/_catalogue.scss:
```diff
@@ -0,0 +1,583 @@
+            display: none;
+          }
+
+          #slide1:checked ~ .slides-wrapper .slide:first-child::before,
```
What is going on here? 
1. I'm not sure you need this selector. 
2. If you do, use a Sass loop for it, DRY

ablyazov-dariy commented on src/styles/_catalogue.scss:
This file has issues; try relying more on inheritance and try to reuse code.
Use classes, and don't nest too deep; it boosts your specificity and looks ugly 
![Uploading Screenshot 2025-03-09 142726.png…]()

ablyazov-dariy commented on src/js/key-navi-catalogue.js:
```diff
@@ -0,0 +1,140 @@
+//  export default
+export default function KeyNaviCatalogue() {
```
Do not use default exports. Function names should be camelCase.

ablyazov-dariy commented on src/js/key-navi-catalogue.js:
```diff
@@ -0,0 +1,140 @@
+}
+
+// others
+export function sidebarNaviCatalogue() {
```
Each function should be in its dedicated file.

ablyazov-dariy commented on src/js/key-navi-catalogue.js:
```diff
@@ -0,0 +1,140 @@
+export default function KeyNaviCatalogue() {
+  const inputs = document.getElementsByTagName("input");
+  const slides = Array.from(inputs).filter(
+    (input) => input.type === "radio" && input.name === "slider"
```
Instead of retrieving all input elements and then filtering for radio buttons with the name 'slider', directly query the required elements using `document.querySelectorAll('input[type="radio"][name="slider"]')`. This approach is more efficient because it reduces the number of elements the browser needs to process, leading to improved performance, especially on pages with a large number of input elements. It also enhances code readability by clearly expressing the intended selection.

ablyazov-dariy commented on src/js/key-navi-catalogue.js:
```diff
@@ -0,0 +1,140 @@
+  const slides = Array.from(inputs).filter(
+    (input) => input.type === "radio" && input.name === "slider"
+  );
+  let currentSlide = Array.from(slides).findIndex((slide) => slide.checked);
```
The `Array.from(slides)` call is unnecessary as `slides` is already an array. Directly use `slides.findIndex((slide) => slide.checked)`. Consider renaming `currentSlide` to `currSlideIdx` for improved clarity. `curr` and `idx` are widely understood abbreviations.

GertSer committed:
Fixed parallax effect + json format

GertSer committed:
Refactored func

GertSer committed:
One more

GertSer committed:
Refactored functions:
key-navi-catalogue.js modal-handler.js open-burger-menu.js parallax.js scroll.js sidebar-navi-blog.js  sidebar-navi-catalogue.js  sidebar.js
